src/lib/machines/extraction/extractionMachine.ts

```diff
@@ -123,12 +134,13 @@ export const extractionMachine = createMachine<ExtractionContext, ExtractionEven
           request: {
             workNumber: ctx.workNumber,
             labwareType: 'Tube',
-            barcodes: ctx.labwares.map((lw) => lw.barcode)
+            barcodes: ctx.labwares.map((lw) => lw.barcode),
+            equipmentId: ctx.equipmentId > 0 ? ctx.equipmentId : undefined
           }
         });
       },
       validateExtraction: (ctx: ExtractionContext) => (send) => {
-        const isValid = ctx.labwares.length > 0 && ctx.workNumber !== '';
+        const isValid = ctx.labwares.length > 0 && ctx.workNumber !== '' && ctx.equipmentId !== 0;
```

Comment:
It would be better to use ctx.equipmentId>0, as in the request , it considers equipmentId > 0 as valid